react/components/SearchBar/components/AutocompleteInput.js

```diff
@@ -1,14 +1,16 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import classNames from 'classnames'
+import { path } from 'ramda'
 
 import { Input } from 'vtex.styleguide'
 import { IconClose, IconSearch } from 'vtex.store-icons'
+import { withRuntimeContext } from 'vtex.render-runtime'
 
 import styles from '../styles.css'
 
 /** Midleware component to adapt the styleguide/Input to be used by the Downshift*/
-export default class AutocompleteInput extends Component {
+class AutocompleteInput extends Component {
```

Comment:
I would definitely recommend changing it to a functional component and use the hook `useRuntime`. Usage of HOCs makes the component tree look bulkier.
Buuuut, i don't think that this component should care about mobile or desktop devices. Should be a concern of the components that make use of it.

Reply:
The `DownsiftComponent` does not work when using a function component, already tried it.

Reply:
Strange... But I believe if you say so.